packages/xo-web/src/common/editable/index.js

```diff
@@ -341,74 +332,87 @@ export class Number extends Component {
-      props.options !== this.props.options
-    ) {
-      this.setState({
-        valueKey: findKey(props.options, option => option === props.value),
-      })
-    }
+    value: PropTypes.oneOfType([PropTypes.oneOf([null]), PropTypes.object]),
   }
 
   get value() {
-    return this.props.options[this.state.valueKey]
-  }
-
-  _onChange = event => {
-    this.setState({ valueKey: getEventValue(event) }, this._save)
+    return this.state.value
   }
 
-  _optionToJsx = (option, key) => {
-    const { renderer } = this.props
+  _onChange = value => this.setState({ value }, value && this._save)
 
+  _renderDisplay() {
+    const { children, renderer, value } = this.props
     return (
-      <option key={key} value={key}>
-        {renderer ? renderer(option) : option}
-      </option>
+      children || (
+        <span>{renderer !== undefined ? renderer(value) : value}</span>
+      )
     )
   }
 
-  _onEditionMount = ref => {
-    // Seems to work in Google Chrome (not in Firefox)
-    ref && ref.dispatchEvent(new window.MouseEvent('mousedown'))
-  }
+  _renderEdition = () => (
+    <FormSelect
+      {...this.props}
+      autoFocus
+      onBlur={this._closeEdition}
+      onChange={this._onChange}
+      onKeyDown={this._onKeyDown}
+      openOnFocus
+    />
+  )
+}
 
-  _renderDisplay() {
-    const { children, renderer, value } = this.props
+class MultiSelect_ extends Editable {
+  static propTypes = {
+    renderer: PropTypes.func,
+    value: PropTypes.oneOfType([
+      PropTypes.oneOf([null]),
+      PropTypes.array,
+      PropTypes.object,
+    ]),
+  }
+  state = { nextValue: this.props.value }
 
-    return children || <span>{renderer ? renderer(value) : value}</span>
+  get value() {
+    return this.state.nextValue === undefined
+      ? this.props.value
+      : this.state.nextValue
   }
 
-  _renderEdition() {
-    const { saving, valueKey } = this.state
-    const { options } = this.props
+  _onChange = nextValue => this.setState({ nextValue })
 
+  _renderDisplay() {
+    const { children, renderer, value } = this.props
     return (
-      <select
-        autoFocus
-        className={classNames('form-control', styles.select)}
-        onBlur={this._closeEdition}
-        onChange={this._onChange}
-        onKeyDown={this._onKeyDown}
-        readOnly={saving}
-        ref={this._onEditionMount}
-        value={valueKey}
-      >
-        {map(options, this._optionToJsx)}
-      </select>
+      children || (
+        <span>{renderer !== undefined ? renderer(value) : value}</span>
+      )
     )
   }
+
+  _onBlur = () => {
+    this._save()
+    this.setState({ nextValue: undefined })
+  }
+
+  _renderEdition = () => (
+    <FormSelect
+      {...this.props}
+      autoFocus
+      onBlur={this._onBlur}
+      onChange={this._onChange}
+      openOnFocus
+      value={this.state.nextValue}
+    />
+  )
 }
 
+export const Select = props =>
+  props.mutli ? <MultiSelect_ {...props} /> : <SimpleSelect_ {...props} />
```

Comment:
No need to pass `multi` to the sub components so use a rest property (`({ multi, ...props }) =>`).